### PR TITLE
Use ArrayPool for Discovery Deserialization

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Alas need to use `ArraySegment<byte>` rather than `Span<byte>` as the DotNetty types don't accept Span

Mild allocation

image

#### What types of changes does your code introduce?

- [x] Optimization

#### Requires testing

- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Core/Buffers/ArrayPoolDisposableReturn.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Buffers;
+
+namespace Nethermind.Core.Buffers;
+
+public readonly struct ArrayPoolDisposableReturn : IDisposable
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Core/Buffers/ArrayPoolDisposableReturn.cs:9]: why not use ArrayPoolList?

**Review by LukaszRozmej — COMMENTED** [review]
Move to ArrayPoolList (add something to it if needed), we don't need more types.

File: src/Nethermind/Nethermind.Core/Buffers/ArrayPoolDisposableReturn.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Core/Buffers/ArrayPoolDisposableReturn.cs:9]: Just wanted type so I could do 

```
using v = ;
```
As single line without either adding a `ArrayPool<byte>.Shared.Return` before each return; or going full try/finally and changing the indentation 😅

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Core/Buffers/ArrayPoolDisposableReturn.cs:9]: ArrayPoolList has that!

**Pull request merged**